src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> newNode = newNode(key, value);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = table[index];
+        while (currentNode.next != null && !Objects.equals(key, currentNode.key)) {
+            currentNode = currentNode.next;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            currentNode.value = value;
+            return;
+        }
+        currentNode.next = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[hash(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        final Node<K, V>[]oldTab = table;
+        table = new Node[table.length * 2];
+        size = 0;
+        for (Node<K, V> node : oldTab) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private Node<K, V> newNode(K key, V value) {
+        return new Node<>(key, value, null);
+    }
+
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
```

Comment:
it should be private constructor, to protect this class from using outside

Reply:
Sorry, my bad))